app/code/Magento/Backend/Controller/Adminhtml/Noroute/Index.php

```diff
@@ -39,4 +39,14 @@ public function execute()
         $resultPage->addHandle('adminhtml_noroute');
         return $resultPage;
     }
+
+    /**
+     * Error page should be public accessible. Do not check keys to avoid redirect loop
+     *
+     * @return bool
+     */
+    protected function _validateSecretKey()
```

Comment:
This can be done in more consistent with other controllers way.

`Magento\Backend\App\AbstractAction` already provides a mechanism for disabling secret key validation: `\Magento\Backend\App\AbstractAction::$_publicActions`. Setting this property will be more correct and consistent fix:

```php
$_publicActions = ['index']
```

However, both solution is fixing the only symptom of an issue, not a root cause. Secret check failed because of invalid value for action (key generated for `denied` and checked for `index`). That should be investigated and fixed as well. Another one issue is why request redirected to `*/*/denied` path which actually does not match any action controller class and fall back to `noroute` controller which does not display own content but has special handling in `Magento\Backend\App\AbstractAction` to render `adminhtml_denied`. In addition `Magento\Backend\Controller\Adminhtml\Denied` action controller is not in proper place and seems should be moved to `Magento\Backend\Controller\Adminhtml\Index` namespace.

As full fix of the issue require deep refactoring of `Magento_Backend` module this fix will be accepted. We will be thankful if you continue your work and fix legacy implementation to achieve more straightforward and predictable routing in admin.